server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestGetMappingAction.java

```diff
@@ -129,47 +131,15 @@ public RestResponse buildResponse(final GetMappingsResponse response, final XCon
                         status = RestStatus.OK;
                     } else {
                         status = RestStatus.NOT_FOUND;
```

Comment:
what do think about moving all this logic to the corresponding transport action (as a follow-up)? I didn't look too close but I think that this causes different behaviour between REST and transport at the moment, it would be nice to have a simpler REST action maybe.

Reply:
I am not talking about removing this (I agree it's not nice and we should talk about that), but rather moving this logic to the transport action. That would maintain the same behaviour at REST and most likely change it for transport client users. Anyway, doesn't belong here.